test/tests/functional/pbs_maintenance_reservations.py

```diff
@@ -440,36 +440,53 @@ def test_maintenance_and_reservation_not_overlap2(self):
                     'reserve_substate': 2}
         self.server.expect(RESV, exp_attr, id=rid1)
 
+    def get_reg_expr(self, _mom, _host):
+        """
+        This function forms the match string based on cpuset mom
+        """
+        if _mom.is_cpuset_mom():
+            n = self.server.status(NODE)
+            cpuset_nodes = [i['id'] for i in n if i['Mom'] == _mom.hostname]
+            reg_str = []
+            if (len(cpuset_nodes) - 1) > 1:
+                for i in range(0, len(cpuset_nodes)-1):
+                    host = _host + '\[%s\]' % i
+                    match_str = '\(%s:ncpus=[0-9]+\)' % host
+                    reg_str.append(match_str)
+                reg_str = ('\+').join(reg_str)
+            else:
+                reg_str = '\(%s:ncpus=[0-9]+\)' % (_host + '\[0\]')
+        else:
+            reg_str = "\(%s:ncpus=[0-9]+\)" % _host
+        return reg_str
+
     @requirements(num_moms=2)
     def test_maintenance_two_hosts(self):
         """
         Test if the maintenance reservation is confirmed on multiple hosts.
         Test the crafted resv_nodes, select, and place.
         Two moms (-p "servers=M1,moms=M1:M2") are needed for this test.
         """
-        if len(self.moms) != 2:
-            cmt = "need 2 mom hosts: -p servers=<m1>,moms=<m1>:<m2>"
-            self.skip_test(reason=cmt)
-
-        now = int(time.time())
-
-        self.server.manager(MGR_CMD_SET, SERVER,
-                            {'managers': '%s@*' % TEST_USER})
-
         self.momA = self.moms.values()[0]
         self.momB = self.moms.values()[1]
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+        reg_expr_hostA = self.get_reg_expr(self.momA, self.hostA)
+        reg_expr_hostB = self.get_reg_expr(self.momB, self.hostB)
 
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'managers': '%s@*' % TEST_USER})
+        now = time.time()
```

Comment:
You're going to want to int() time.time() since it returns a float including microseconds.